### ISSUE TITLE
using cookie from wrong account

### ISSUE THREAD

**Issue by danksky** [costastf/locationsharinglib#98]
I have 2 accounts signed in on my browser:

a@gmail.com and b@gmail.com

I signed into both manually, and replaced my cookies.txt file

```python
from locationsharinglib import Service
cookies_file = '/somepath/cookies.txt'
google_email = 'b@gmail.com'
service = Service(cookies_file=cookies_file,
                  authenticating_account=google_email)
for person in service.get_all_people():
    print(person)
```

For some reason, when I run the above code, it grabs only the people that are shared with a@gmail.com, even though `google_email` is set to "b@gmail.com". 

_This was working before,_ but it stopped working when I did this:
1. I signed out of all accounts on Chrome (invalidating the cookie)
1. I cleared all cookies from my browser 
1. I signed back in

It seems as if `authenticating_account` doesn't actually check that the account value matches the cookie?

**Comment by costastf** [comment]
Hi @danksky, you are right the library does no matching of cookies per email provided, nor can it, the cookies are just provided through the session and the backend service picks up whatever it likes. The email provided is just for presentation reasons. There was never the intention for the library to support two accounts but only to support one with other accounts shared to it. I don't think there is anything we can do about this usage case, I am open to suggestions though. I hope this makes sense and thanks for reporting!

**Comment by costastf** [comment]
Hi @danksky , do you have any thoughts or feedback or shall we close this issue?

**Comment by danksky** [comment]
Hi @costastf, thanks for the quick response! I appreciate the explanation.
I would say, if anything, this calls for improved documentation as to what
the argument is for.
The cookie-acquisition process is definitely the trickiest part of the
usage of this library, so a guide would be VERY helpful. Of course, more
documentation increases the surface area of what needs to be updated as the
cookie acquisition process changes.

For posterity, I used Brave to generate the cookie. It’s basically Chrome,
but with more privacy on the web built in. I only signed into the account
that I was planning to use this library for. I use plain Chrome for
everything else.

There are also a few logistical questions about the cookie that would help
to provide answers for, maybe in an FAQ section or another section in the
READMe dedicated to the cookie.

- how long does the cookie last
- can the cookie be provided as a string, and if so, how
- how to interpret the cookie from Cookies.txt and why it’s different from
the file exported using other extensions
- why that extension is recommended

What do you think?

**Comment by costastf** [comment]
Hi @danksky ,
All your comments make a ton of sense. The cookie gathering process was initially an automatic process https://github.com/costastf/locationsharinglib/blob/b518c89bc37dbec113c94f6fb80c86b0fe7d300a/locationsharinglib/locationsharinglib.py#L233, later it was a guided process https://github.com/costastf/mapscookiegettercli and lastly it was implemented as it is now with providing the cookies through an external process. 

The documentation is very thin admitently so that definitely can be improved althouth there are very long threads on home assistant project (for example https://community.home-assistant.io/t/google-maps-location-sharing-new-thread/54555/180)  about it so i guess that most people manage through that. That being said it would be nice for everybody if the documentation was improved. The truth is that this is one of the many many libraries that I maintain and I could definitely use the help so if you would like to help with documentation improvement it would be very much appreciated.

As for the questions mentioned,

- how long does the cookie last
    The cookie does not expire as far as i know, but of course that means that one would need to "burn" a session to a browser and never close that so the cookie does not get invalidated. I guess what happens is that people do not follow the suggestion of creating a new account just for the sharing, they reuse one of their actual accounts and at some point they log out of a session, killing the cookie that has been created with that session.
    
- can the cookie be provided as a string, and if so, how
    Not really sure why is that relevant at all and how it would make any difference or be implemented, maybe you can elaborate a bit on it

- how to interpret the cookie from Cookies.txt and why it’s different from the file exported using other extensions
    Basically what is needed is an extension that provides curl compatible cookie dump. Maybe it would be a good exercise to try out a few extensions and mention the ones that work. 
    
- why that extension is recommended
    would probably be covered by the previous topic.
    
    
I hope the above makes some sense, i am a little sleep deprived :)

**Comment by costastf** [comment]
Hi @danksky , any thoughts on this thread or should I close this issue?

**Comment by danksky** [comment]
My main concern was about the cookie expiring. Glad to know there is
technically no limit. Thanks!
